Formula/dotnet.rb

```diff
@@ -22,10 +17,10 @@ class Dotnet < Formula
   depends_on "pkg-config" => :build
   depends_on xcode: :build
   depends_on arch: :x86_64
-  depends_on "curl"
-  depends_on "icu4c"
   depends_on "openssl@1.1"
 
+  uses_from_macos "curl"
+  uses_from_macos "icu4c"
```

Comment:
I don't know, @carlocab. Are you suggesting replacing the two lines with the following?

```suggestion
  uses_from_macos "libicucore"
```

Reply:
No, not quite. That replacement would lead to an error. My point is that the `icu4c` formula provides a lot more than the version that comes with macOS, which is why I ask whether dotnet doesn't need all the extra bits.

Reply:
I can try to remove `icu4c` and see whether the build succeeds? Your comment is for both line 22 and 23, should I do anything about `curl` as well?

```suggestion
```

Reply:
It will likely fail, since `libicu` is listed as a dependency: https://github.com/dotnet/runtime/blob/main/docs/workflow/requirements/linux-requirements.md#toolchain-setup

We can try dropping the `curl` dependency.

Reply:
I see Fedora dropped `libcurl`, so may be safe to entirely remove `curl` dependency from both macOS and Linux.